Assets/MixedRealityToolkit/Utilities/Async/Internal/AsyncCoroutineRunner.cs

```diff
@@ -75,10 +83,17 @@ internal static AsyncCoroutineRunner Instance
                 }
 
                 instance.gameObject.hideFlags = HideFlags.None;
-#if !UNITY_EDITOR
```

Comment:
No longer needed?

Reply:
This was a question asked here: https://github.com/microsoft/MixedRealityToolkit-Unity/issues/4988#issuecomment-503968696.

I was searching through history and also on the upstream thing (i.e. the original source of the code that we forked from), and what I found was:

1. There was no history/rationale/comments/reasoning associated with the UNITY_EDITOR change.
2. The original code didn't have this check (i.e. it always did DontDestroyOnLoad in both editor and player.

I opted to be consistent with upstream here absent any known/compelling reason to be inconsistent (also in general, it feels like having the same code run in editor/player is a good thing to ensure that testing in-editor leads to the same code being tested in player).

Reply:
I think there may have been some side effect of calling this at edit-time. Not sure if the work done since then prevents this code from being called then, so it may no longer be necessary.